thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java

```diff
@@ -47,20 +41,64 @@ public OnboardResource() {
 
   public OnboardResource(AnomalyFunctionManager anomalyFunctionManager,
       MergedAnomalyResultManager mergedAnomalyResultManager,
-      RawAnomalyResultManager rawAnomalyResultManager){
+      RawAnomalyResultManager rawAnomalyResultManager) {
     this.anomalyFunctionDAO = anomalyFunctionManager;
     this.mergedAnomalyResultDAO = mergedAnomalyResultManager;
     this.rawAnomalyResultDAO = rawAnomalyResultManager;
   }
 
-  // endpoint clone function Ids to append a name defined in nameTags
+ // Useful end points to check for function information
 
   @GET
   @Path("function/{id}")
   public AnomalyFunctionDTO getAnomalyFunction(@PathParam("id") Long id) {
     return anomalyFunctionDAO.findById(id);
   }
 
+  @POST
+  @Path("function/activate")
+  public String activateFunction(@QueryParam("functionIds") String functionIds) {
+    toggleFunctions(functionIds, true);
+    return functionIds;
+  }
+
+  @POST
+  @Path("function/deactivate")
+  public String deactivateFunction(@QueryParam("functionIds") String functionIds) {
+    toggleFunctions(functionIds, false);
+    return functionIds;
+  }
+
+  /**
+   * toggle anomaly functions to active and inactive
+   *
+   * @param functionIds string comma separated function ids, if
+   * @param tag boolean true or false, set function as true or false
+   */
+  private void toggleFunctions(String functionIds, boolean tag) {
+    List<Long> functionIdsList = new ArrayList<>();
+    if (functionIds.equals("ALL")) {
```

Comment:
During the experiment process I found myself using these activate/deactivate function so many times. Its essentially just of way communicating with the application database. Java do not support good 'cli' tools that's why I prefer endpoints over directly invoke the toggle function.